### ISSUE TITLE
Writing generated doc to an output stream does not work (undefined method `<<')

### ISSUE THREAD

**Issue by mpe85** [asciidoctor/asciidoctorj#588]
If I try to write the generated document (in this case with backend PDF) to an output stream I get this exception:

```
Exception in thread "main" org.asciidoctor.internal.AsciidoctorCoreException: org.jruby.exceptions.RaiseException: (NoMethodError) undefined method `<<' for #<Java::JavaIo::ByteArrayOutputStream:0x7378ff35>
	at org.asciidoctor.internal.JRubyAsciidoctor.convert(JRubyAsciidoctor.java:460)
	at org.asciidoctor.internal.JRubyAsciidoctor.convert(JRubyAsciidoctor.java:486)
	at org.asciidoctor.internal.JRubyAsciidoctor.convert(JRubyAsciidoctor.java:481)
	at Test.main(Test.java:82)
Caused by: org.jruby.exceptions.RaiseException: (NoMethodError) undefined method `<<' for #<Java::JavaIo::ByteArrayOutputStream:0x7378ff35>
	at RUBY.render_header(uri:classloader:/gems/pdf-core-0.7.0/lib/pdf/core/renderer.rb:191)
	at RUBY.render(uri:classloader:/gems/pdf-core-0.7.0/lib/pdf/core/renderer.rb:164)
	at RUBY.render(uri:classloader:/gems/prawn-2.2.2/lib/prawn/document.rb:378)
	at RUBY.write(uri:classloader:/gems/asciidoctor-pdf-1.5.0.alpha.16/lib/asciidoctor-pdf/converter.rb:2826)
	at RUBY.write(uri:classloader:/gems/asciidoctor-1.5.6.1/lib/asciidoctor/document.rb:1083)
	at RUBY.convert(uri:classloader:/gems/asciidoctor-1.5.6.1/lib/asciidoctor.rb:1507)
	at RUBY.convert(<script>:85)
```

I call asciidoctorj like this:
```
ByteArrayOutputStream baos = new ByteArrayOutputStream();
Asciidoctor.Factory.create().convert(
		out,
		OptionsBuilder.options()
				.backend("pdf")
				.toStream(baos)
				.headerFooter(true)
				.safe(SafeMode.UNSAFE)));
```
The same happens for each other implementation of OutputStream.

Environment:
asciidoctorj 1.6.0 alpha 6
asciidoctorj-pdf 1.5.0 alpha 16
jruby 9.1.13.0

**Comment by robertpanzer** [comment]
I can confirm the error.
It also happens with the html5 backend with another error though.
I guess we need a bridge that we pass to Asciidoctor.

**Comment by robertpanzer** [comment]
Asciidoctor core seems to be aware of streams and checks if there is a method 'write' to invoke.
But JRuby can't figure out which one to invoke for a java.io.OutputStream:
```
Caused by: org.jruby.exceptions.RaiseException: (NameError) no method 'write' for arguments (org.jruby.RubyString) on Java::JavaIo::ByteArrayOutputStream
  available overloads:
    (int)
    (byte[])
	at RUBY.write(uri:classloader:/gems/asciidoctor-1.5.6.1/lib/asciidoctor/document.rb:1087)
```

I think we have to provide a wrapper that implements `write(String)` called by Asciidoctor and `<<(String)` called by pdf-core.

**Comment by robertpanzer** [comment]
Fixed with #592
